src/utils/metrics.cpp

```diff
@@ -24,8 +24,10 @@
 
 namespace dsn {
 
-metric_entity::metric_entity(const std::string &id, attr_map &&attrs)
-    : _id(id), _lock(), _attrs(std::move(attrs)), _metrics()
+metric_entity::metric_entity(const metric_entity_prototype *prototype,
+                             const std::string &id,
+                             const attr_map &attrs)
+    : _prototype(prototype), _id(id), _lock(), _attrs(attrs), _metrics()
```

Comment:
How about let _lock and _metrics construct implicitly and automatically?

Reply:
OK.